src/utils/RequestQueue.ts

```diff
@@ -30,13 +30,24 @@ interface RequestItem<V> {
  * while the original request is still in the queue.
  */
 export default class RequestQueue {
-  /** The maximum number of requests that can be handled concurently.
-    Once reached, additional requests will be queued up to run once a running request completes.*/
+  /**
+   * The maximum number of requests that can be handled concurently.
+   * Once reached, additional requests will be queued up to run once a running request completes.
+   */
   private maxActiveRequests: number;
 
+  /**
+   * The maximum number of requests that can be handled concurrently for low-priority requests to be queued. Equal to
+   * `maxActiveRequests` by default, but may be set lower to ensure the queue leaves space for high-priority requests.
```

Comment:
seems like the default should always be to allow high priority requests and keep this one more conservative?